chirp/drivers/radtel_rt493.py

```diff
@@ -0,0 +1,1152 @@
+    return bytes.fromhex(hex_str)[::-1]
+
+
+def _decode_freq(hex_bytes):
+    """
+    Decodes a 4-byte frequency representation into MHz.
+    The bytes must be read **right to left** in HEX format (LBCD):
+        Byte3 Byte2 . Byte1 Byte0
+    Example:
+        00 50 13 43  ->  431.350 MHz
+    """
+    b0, b1, b2, b3 = hex_bytes
+
+    if b0 == 0xFF and b1 == 0xFF and b2 == 0xFF and b3 == 0xFF:
+        return 0.0
+
+    # Extract the high and low nibbles (hex digits) of b2
+    b2_high = (b2 >> 4) & 0xF  # Get the first hex digit (upper nibble)
+    b2_low = b2 & 0xF          # Get the second hex digit (lower nibble)
+
+    form = f'{b3:02X}{b2_high:X}{b2_low:X}{b1:02X}{b0:02X}'
+    return int(form)
+
+
+def _decode_channel_settings(byte):
+    """
+    Decodes the main channel settings byte into a structured dictionary.
+
+    :param byte: The byte to decode (integer)
+    :return: Dictionary with settings
+    """
+    return {
+        # (1=Wide, 0=Narrow)
+        "W/N Mode": "Wide" if ord(byte) & 0x10 else "Narrow",
+        "Compander": bool(ord(byte) & 0x80),  # (1=ON, 0=OFF)
+        "Scramble Mode": ord(byte) & 0x07,  # (0=OFF, 1-7=Scramble levels)
+        "Special QT/DQT": "A" if ord(byte) & 0x20 else
+        ("B" if ord(byte) & 0x40 else "OFF"),
+    }
+
+
+def _encode_channel_settings(wn, scramble, compander, special_qt_dqt):
+    """
+    Encodes the main channel settings into a single byte.
+
+    :param wn: "wide" or "narrow"
+    :param scramble: Scramble mode (0 = OFF, 1-8 = Scramble levels)
+    :param compander: Boolean, True = ON, False = OFF
+    :param special_qt_dqt: "off", "A", "B"
+
+    :return: Encoded settings byte (integer)
+    """
+    settings_byte = 0x00  # Start with 0000 0000
+
+    # Wide/Narrow: 1 = Wide, 0 = Narrow
+    if wn.lower() == "wide":
+        settings_byte |= 0x10
+
+    # Compander: 1 = ON, 0 = OFF
+    if compander:
+        settings_byte |= 0x80
+
+    # Scramble Mode
+    if scramble > 0:
+        settings_byte |= (scramble & 0x07)
+
+    # Special QT/DQT: 0 = OFF, A = 1, B = 2
+    if special_qt_dqt.upper() == "A":
+        settings_byte |= 0x02
+    elif special_qt_dqt.upper() == "B":
+        settings_byte |= 0x06
+
+    return settings_byte.to_bytes(1, "big")
+
+
+def do_download(radio):
+    LOG.debug("Downloading data from radio")
+
+    status = chirp_common.Status()
+    status.msg = "Downloading from radio"
+
+    status.cur = 0
+    status.max = radio._upper
+    radio.status_fn(status)
+
+    data = bytearray()
+
+    _enter_programming_mode(radio)
+    for i in range(1, radio.MEM_ROWS, 2):
+        status.cur = i
+        radio.status_fn(status)
+
+        # block = radio._cache_block(i, in_prog_mode=True)
+        result = _read_block(radio, i)
+        # block1, block2 = result[:16], result[16:]
+        data.extend(result)
+
+        LOG.debug("Downloaded memory channel %i" % i)
+        # LOG.debug(block)
+
+    _exit_programming_mode(radio)
```

Comment:
This should be in a `finally` block to make sure it runs even if there is an error. Easiest way to do that is in your `sync_in()` and `sync_out()` methods.